code/game/machinery/machine_frame.dm

```diff
@@ -279,7 +279,9 @@
 		return .
 	if(state != FRAME_STATE_BOARD_INSTALLED)
 		return .
-
+	if(locate(circuit.build_path) in loc)
+		balloon_alert(user, "identical machine present!")
+		return ITEM_INTERACT_BLOCKING
```

Comment:
Should be in `finalize_construction`

Reply:
okay i didnt know if i should have it before or after the component check, i figured before so it tells the player when they are clearly trying to build a machine when they cant, regardless if they have the parts or not.